### PR TITLE
Refactor signer pruning with upkeep service

### LINKED ISSUE #2075: Refactor pruning with upkeep service in signer/aggregator
## Why
We need to remove the pruning from the save functions of the `StakePoolStore` and `EpochSettingsStore` and implement it in the upkeep service in the aggregator and signer.

## What
Migrate pruning of the `StakePoolStore` and `EpochSettingsStore` to the upkeep service in signer and aggregator.

## How

- [x] Replicate the upkeep service [logic](https://github.com/input-output-hk/mithril/blob/main/mithril-signer/src/services/upkeep_service.rs#L33) implemented in the signer to the aggregator
- [x] Implement the `EpochPruningTask` for the  `StakePoolStore` and `EpochSettingsStore`  in:
  - [x] the signer
  - [x] the aggregator

### PR DESCRIPTION
**Pull request by sfauvel**
## Content

This PR remove the pruning from the save function and use the upkeep service instead.
The `StakeStore` is moved from `persistence` to `signer`.
The `EpochPruningTask` is implemented over the `StorePruner` in the Signer for the `ProtocolInitializerStore` and `StakeStore`.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)

Closes #2075

### CODE REVIEW

File: mithril-signer/src/store/stake_store.rs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+use mithril_persistence::store::StakeStorer;
+use tokio::sync::RwLock;
+
+use mithril_persistence::store::{adapter::StoreAdapter, StorePruner};
+
+use crate::services::EpochPruningTask;
+
+type Adapter = Box<dyn StoreAdapter<Key = Epoch, Record = StakeDistribution>>;
+
+/// A [StakeStorer] that use a [StoreAdapter] to store data.
```
Comment (jpraynaud) [inline mithril-signer/src/store/stake_store.rs:14]:
```suggestion
/// A [StakeStorer] that uses a [StoreAdapter] to store data.
```

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Review by Alenar — APPROVED** [review]
LGTM

**Review by dlachaume — APPROVED** [review]
LGTM 🚀

File: mithril-signer/src/store/stake_store.rs (hunk 1 above)
Comment (sfauvel) [reply mithril-signer/src/store/stake_store.rs]: Done

**Pull request merged**